File: frontend/src/types/notification.ts
export interface NotificationChannel {
  id: string;
  name: string;
  apprise_url: string;
  enabled: boolean;
  created_at: string;
  updated_at: string;
  task_count?: number;
}

export interface CreateChannelPayload {
  name: string;
  apprise_url: string;
  enabled: boolean;
}

File: backend/src/models/notification.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

/// Vue enrichie pour la liste (avec le nombre de tâches référençant ce canal)
#[derive(Debug, Serialize)]
pub struct ChannelWithTaskCount {
    pub id: Uuid,
    pub name: String,
    pub apprise_url: String,
    pub enabled: bool,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub updated_at: chrono::DateTime<chrono::Utc>,
    pub task_count: i64,
}

#[derive(Debug, Deserialize)]
pub struct CreateChannelRequest {
    pub name: String,
    pub apprise_url: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
}

#[derive(Debug, Deserialize)]
pub struct UpdateChannelRequest {
    pub name: String,
    pub apprise_url: String,
    pub enabled: bool,
}

fn default_true() -> bool {
    true
}
